.github/workflows/lint.yml

```diff
@@ -30,7 +30,7 @@ jobs:
     - uses: actions/checkout@v4.1.1
       with:
         fetch-depth: 0  # previous commits used in tools/lint.py
-        submodules: recursive
+        submodules: false
```

Comment:
I see what you mean. Do you think the tools should instead be modified to check for submodule paths? In the case of lint.py it could easily generate a lot of errors against another repo that either doesn't have linting or allows a larger line length, and in both cases it's not something that SciPy can necessarily do anything about. Pretty much the same for unicode-check and check_test_name. I can do that in a separate PR, at which point the value of this variable would be moot and we could either change it back or leave it as is.

Reply:
The issue happened because prima includes pybind11 as a submodule, and their test_builtin_casters fails the unicode-check (and they are indeed checking some unicode conversions) as well as check_test_name. Neither of those have the —diff-against option. Should I make a separate PR to refactor the submodule logic I added to check_installation so that it can be used in unicode-check and check_test_names? (And also can I please rename unicode-check to check_unicode so that it’s consistent with the other checks?)